test/ProfilerTest.cpp

```diff
@@ -45,5 +53,124 @@ TEST(ProfilerTest, CreateForUDP) {
+  socklen_t len = sizeof(cliaddr);
+  int retbuflen = recvfrom(sockfd, buffer, buffer_size, 0,
+                           (struct sockaddr *)&cliaddr, &len);
+  if (retbuflen != buffer_size) return false;
+  if (buffer[0] == 68U && buffer[37] == 71U && buffer[94] == 115U) {
+    close(sockfd);
+    return true;
+  } else {
+    close(sockfd);
+    return false;
+  }
+}
+
+class Human {
+ public:
+  Human(const std::string &name, int age) : name_(name), age_(age) {}
+
+  std::string GetName() const { return name_; }
+
+  int GetAge() const { return age_; }
 
-// TODO Tests using Scheduler
+ private:
+  std::string name_;
+  int age_;
+};
+
+class HumanReverser {
+ public:
+  HumanReverser(pipert::PolledChannel<Human> *pc_to_write)
+      : pc_to_write_(pc_to_write) {}
+
+  void ReverseName(pipert::PacketToProcess<Human> p) {
+    EXPECT_EQ(p.data().GetName(), "Jimi Hendrix");
+    EXPECT_TRUE(p.data().GetAge() >= 28 && p.data().GetAge() <= 37);
+    std::string name(p.data().GetName());
+    std::reverse(name.begin(), name.end());
+    pc_to_write_->Acquire(p.timestamp() + 10, name, p.data().GetAge());
+  }
+
+ private:
+  pipert::PolledChannel<Human> *pc_to_write_;
+};
+
+class HumanPrinter {
+ public:
+  HumanPrinter(pipert::ScheduledChannel<Human> *ch_to_write)
+      : ch_to_write_(ch_to_write) {}
+
+  void PrintName(pipert::PacketToProcess<Human> p) {
+    EXPECT_EQ(p.data().GetName(), "Jimi Hendrix");
+    EXPECT_TRUE(p.data().GetAge() >= 28 && p.data().GetAge() <= 37);
+    // std::cout << packet.data().GetName() << std::endl;
+    ch_to_write_->Acquire(p.timestamp() + 5, p.data());
+  }
+
+ private:
+  pipert::ScheduledChannel<Human> *ch_to_write_;
+};
+TEST(ProfilerTest, TestUDPUsingSchedular) {
+  PrepareToReceive();
+  pipert::Scheduler sch(0, pipert::Profiler("udp:127.0.0.1:8888"));
+  int channel_capacity = 10;
+  pipert::PolledChannel<Human> pc =
+      sch.CreatePolledChannel<Human>("OutChannel", channel_capacity);
+
+  HumanReverser hr(&pc);
+  pipert::ScheduledChannel<Human> sc2 = sch.CreateScheduledChannel<Human>(
+      "ReverserChannel", channel_capacity, nullptr,
+      std::bind(&HumanReverser::ReverseName, &hr, std::placeholders::_1));
+
+  HumanPrinter hp(&sc2);
+  pipert::ScheduledChannel<Human> sc1 = sch.CreateScheduledChannel<Human>(
+      "PrinterChannel", channel_capacity, nullptr,
+      std::bind(&HumanPrinter::PrintName, &hp, std::placeholders::_1));
+
+  sch.Start();
+
+  pipert::Timer::Time time = pipert::Timer::time();
+  for (int i = 0; i < 10; i++) {
+    pipert::PacketToFill<Human> packet_to_fill =
+        sc1.Acquire(time, "Jimi Hendrix", 28 + i);
+  }
+
+  while (pc.GetQueuedBufferLength() != channel_capacity)
+    std::this_thread::yield();
+
+  sch.GetProfiler().GatherNSend();
+
+  sch.Stop();
+
+  for (pipert::PacketToProcess<Human> packet_to_process = pc.Poll();
+       !packet_to_process.IsEmpty(); packet_to_process = pc.Poll()) {
+  }
+  // test receiving data
+  EXPECT_EQ(Receive(), true);
```

Comment:
This line fails in clang release (returns false). It would be good to find out, why.